go/vt/discovery/healthcheck.go

```diff
@@ -136,156 +154,39 @@ func ParseTabletURLTemplateFromFlag() {
-	TabletExternallyReparentedTimestamp int64
-	// Stats is the current health status, as received by the
-	// StreamHealth RPC (replication lag, ...).
-	Stats *querypb.RealtimeStats
-	// LastError is the error we last saw when trying to get the
-	// tablet's healthcheck.
-	LastError error
-}
-
-// String is defined because we want to print a []*TabletStats array nicely.
-func (e *TabletStats) String() string {
-	return fmt.Sprint(*e)
-}
-
-// DeepEqual compares two TabletStats. Since we include protos, we
-// need to use proto.Equal on these.
-func (e *TabletStats) DeepEqual(f *TabletStats) bool {
-	return e.Key == f.Key &&
-		proto.Equal(e.Tablet, f.Tablet) &&
-		e.Name == f.Name &&
-		proto.Equal(e.Target, f.Target) &&
-		e.Up == f.Up &&
-		e.Serving == f.Serving &&
-		e.TabletExternallyReparentedTimestamp == f.TabletExternallyReparentedTimestamp &&
-		proto.Equal(e.Stats, f.Stats) &&
-		((e.LastError == nil && f.LastError == nil) ||
-			(e.LastError != nil && f.LastError != nil && e.LastError.Error() == f.LastError.Error()))
-}
-
-// Copy produces a copy of TabletStats.
-func (e *TabletStats) Copy() *TabletStats {
-	ts := *e
-	return &ts
-}
-
-// GetTabletHostPort formats a tablet host port address.
-func (e TabletStats) GetTabletHostPort() string {
-	vtPort := e.Tablet.PortMap["vt"]
-	return netutil.JoinHostPort(e.Tablet.Hostname, vtPort)
-}
-
-// GetHostNameLevel returns the specified hostname level. If the level does not exist it will pick the closest level.
-// This seems unused but can be utilized by certain url formatting templates. See getTabletDebugURL for more details.
-func (e TabletStats) GetHostNameLevel(level int) string {
-	chunkedHostname := strings.Split(e.Tablet.Hostname, ".")
-
-	if level < 0 {
-		return chunkedHostname[0]
-	} else if level >= len(chunkedHostname) {
-		return chunkedHostname[len(chunkedHostname)-1]
-	} else {
-		return chunkedHostname[level]
-	}
+func init() {
+	// Flags are not parsed at this point and the default value of the flag (just the hostname) will be used.
+	ParseTabletURLTemplateFromFlag()
+	flag.Var(&TabletFilters, "tablet_filters", "Specifies a comma-separated list of 'keyspace|shard_name or keyrange' values to filter the tablets to watch")
+	topoproto.TabletTypeListVar(&AllowedTabletTypes, "allowed_tablet_types", "Specifies the tablet types this vtgate is allowed to route queries to")
+	flag.Var(&KeyspacesToWatch, "keyspaces_to_watch", "Specifies which keyspaces this vtgate should have access to while routing queries or accessing the vschema")
 }
 
-// NamedStatusURL returns the URL for the case where a tablet server is named.
-func (e TabletStats) NamedStatusURL() string {
-	return "/" + topoproto.TabletAliasString(e.Tablet.Alias) + servenv.StatusURLPath()
-}
+// TabletRecorder is the part of the HealthCheck interface that can
+// add or remove tablets. We define it as a sub-interface here so we
+// can add filters on tablets if needed.
+type TabletRecorder interface {
```

Comment:
Is there a reason why this is a separate interface instead of being part of Healthcheck?

Reply:
It is still being used by the old healthcheck and offshoots (Filters)